static/app/views/replays/detail/trace/index.tsx

```diff
@@ -18,10 +20,43 @@ function PerfDisabled() {
   );
 }
 
+type EventHandler = (...args: any[]) => void;
+
+class EventEmitter {
+  private events: {[key: string]: EventHandler[]} = {};
+
+  on(event: string, handler: EventHandler): void {
+    this.events[event] = this.events[event] || [];
+    this.events[event].push(handler);
```

Comment:
With a design like this, it is possible to push the same emitter multiple times and it will result in multiple invocations. Is that expected?